sources/test.py

```diff
@@ -34,6 +34,8 @@
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description=("Convolutional Neural Netw"
                                                   "ork on street-scene images"))
+    parser.add_argument('-a', '--aggregate-label', action='store_true',
```

Comment:
I think, for further pull requests, we can factorize some code between the `train.py` and `test.py` modules because I see multiple duplication: the parsing of the command line, folders and files checking, etc.

Reply:
Be careful, the command line parsing is not managed exactly in the same manner in both files. The point is that we must point to the accurate `<datadir>/<dataset>/output/<problem>/checkpoints/<instance_name>` checkpoint to recover the model. Either we set all arguments as in `train.py`, or we set only one argument, that is the whole `<instance_name>` string. I chose this second way, as some arguments are not useful in `test.py`. Do you think the first way is better?

Reply:
I'm aware that all arguments parsing can not be shared between `train` and `test`. It's just a feeling that some stuff could be factorized when I saw the current diffs. I wonder if it's relevant.

Reply:
You're right, it is relevant. The dataset creation and populating, as well as the glossary management, are common. The only parts that must be differentiated are argument parsing and model calling. Do we address it in this PR?